pkg/compactor/bucket_compactor.go

```diff
@@ -541,6 +541,26 @@ func IsOutOfOrderChunkError(err error) bool {
 	return ok
 }
 
+// CriticalError is a type wrapper for block health critical errors.
+type CriticalError struct {
+	err error
+	id  ulid.ULID
+}
+
+func (e CriticalError) Error() string {
+	return e.err.Error()
+}
+
+func criticalError(err error, brokenBlock ulid.ULID) CriticalError {
+	return CriticalError{err: err, id: brokenBlock}
+}
+
+// IsCriticalError returns true if the base error is a CriticalError.
+func IsCriticalError(err error) bool {
+	_, ok := errors.Cause(err).(CriticalError)
+	return ok
+}
```

Comment:
We should use this pattern instead (same in `IsIssue347Error` and `IsOutOfOrderChunkError`):

```suggestion
func IsCriticalError(err error) (bool, CriticalError) {
	var criticalError CriticalError
	ok := errors.As(err, &criticalError)
	return ok, criticalError
}
```

Reply:
_changed @ 5d2384a3c78460c4363493396161d990d12caee3_